### ISSUE TITLE
Accessibility: Screen Reader Support

### ISSUE THREAD

**Issue by chigkim** [earendil-works/pi#4593]
### What happened?

TUI utilizes bunch of ascii arts characters in borders, dynamic decorations, selector formatting, and footer layout, and it is noisy and confusing when read aloud by screen readers.
For example, screen readers keeps repeating "box drawing light horizontal" bunch of times before and after prompt field.
Also it uses braille unicode for spinners. The characters are supposed to be used for Braille display.
It is hard to find the start of assistant response without visual marking.

### Steps to reproduce

Run `pi` in terminal, and read the UI with VoiceOver on Mac.

### Expected behavior

The cli should have -sr flag to make TUI easier to use with screen readers by enabling a flatter, less decorative rendering mode.
Keep the existing visual TUI behavior unchanged when the flag is not used.
Also add User: and Assistant: to mark the start of message role.
Gemini-cli has this feature.
I have a vibe coded branch as a concept.
https://github.com/chigkim/pi/tree/add-sr-flag
Thanks!

### Version

v0.74.0
